wayland-client/src/event_queue.rs

```diff
@@ -582,6 +622,57 @@ impl<State: 'static> QueueHandle<State> {
 
         Arc::new(QueueProxyData::<I, U> { sender, udata: user_data, _phantom: PhantomData })
     }
+
+    /// Temporarily block processing on this queue.
+    ///
+    /// This will cause the associated queue to appear empty until all [`FreezeHandle`]s associated
+    /// with the queue are dropped.
+    pub fn freeze(&self) -> FreezeHandle<State> {
+        self.inner.lock().unwrap().freeze_count += 1;
+        FreezeHandle { qh: self }
+    }
+}
+
+impl<'a, State: 'static> FreezeHandle<'a, State> {
+    /// Create a placeholder object data associated with this event queue.
+    ///
+    /// This is needed when creating composite objects whose user data cannot be produced until all
+    /// components have been created, which would normally produce a cyclic dependency.  If this
+    /// data is not replaced by an object data obtained from [`QueueHandle::make_data`] with
```

Comment:
I'm assuming that "replaced" part is relating to `Backend::set_data`? As discussed on Matrix I'm thinking this method probably should be removed, as it is quite fragile and racy now that the backend has internal locking. So I'm rather uncertain about this.

Reply:
If `set_data` is removed, then `make_early_data` has no use, but FreezeHandle can still be useful.

Reply:
Yeah, I'm thinking we need some kind of mechanism to allow for late-init of the user-data, I'm just wondering if we can manage to do something more robust :thinking: